api/oneclick.py

```diff
@@ -155,16 +170,157 @@ def circulate_item(self, patron_id, item_id, return_item=False):
+
+        resp_dict = self.circulate_item(patron_id=patron_oneclick_id, item_id=item_oneclick_id, hold=True, return_item=False)
+
+        if not resp_dict or ('error_code' in resp_dict):
+            return None
+
+        self.log.debug("Patron %s/%s checked out item %s with transaction id %s.", patron.authorization_identifier, 
+            patron_oneclick_id, item_oneclick_id, resp_dict['transactionId'])
+
+        today = datetime.datetime.now()
+
+        hold = HoldInfo(
+            identifier_type=licensepool.identifier.type,
+            identifier=item_oneclick_id,
+            start_date=today,
+            # OneClick sets hold expirations to 2050-12-31, as a "forever"
+            end_date=None,
+            hold_position=0,
+        )
+
+        return hold
+
+
+    def release_hold(self, patron, pin, licensepool):
+        """Release a patron's hold on a book.
+
+        :param patron: a Patron object for the patron who wants to return the book.
+        :param pin: The patron's password (not used).
+        :param licensepool: The Identifier of the book to be checked out is 
+        attached to this licensepool.
+
+        :return True on success, raises circulation exceptions on failure.
+        """
+        patron_oneclick_id = self.validate_patron(patron)
+        (item_oneclick_id, item_media) = self.validate_item(licensepool)
+
+        resp_dict = self.circulate_item(patron_id=patron_oneclick_id, item_id=item_oneclick_id, hold=True, return_item=True)
+
+        if resp_dict == {}:
+            self.log.debug("Patron %s/%s released hold %s.", patron.authorization_identifier, 
+                patron_oneclick_id, item_oneclick_id)
+            return True
+
+        # should never happen
+        raise CirculationException("Unknown error %s/%s releasing %s.", patron.authorization_identifier, 
+            patron_oneclick_id, item_oneclick_id)
+
+
+
+    ''' -------------------------- Patron Account Handling -------------------------- '''
+    def create_patron(self, patron):
+        """ Ask OneClick to create a new patron record.
+
+        :param patron: a Patron object which contains the permanent id to give to 
+        OneClick as the patron's library card number.
+        :return OneClick's internal patron id.
+        """
+
+        patron_cardno = patron.authorization_identifier
+        if not patron_cardno:
+            raise InvalidInputException("Patron %r has no card number.", patron)
+
+        url = "%s/libraries/%s/patrons/" % (self.base_url, str(self.library_id))
+        action="create_patron"
+        
+        args = dict()
+        args['libraryId'] = self.library_id
+        args['libraryCardNumber'] = patron.authorization_identifier
+        # generate indicative values for the account fields the patron has not supplied us with
+        args['userName'] = 'username' + patron.authorization_identifier
+        args['email'] = 'patron' + patron.authorization_identifier + '@librarysimplified.org'
```

Comment:
Actually, just generate UUIDs and use them for username and email. There's no reason to give One Click any patron information at all here.